### DISCUSSION TITLE
Desktop file entries go to Lost and Found on KDE

### DISCUSSION THREAD

**Discussion by bkmo** [filips123/PWAsForFirefox#180] [Ideas]
On KDE Plasma there are no menu categories as Categories=GTK;WebApps;;
Maybe adding Network  would allow it to go into the Internet section of the menu.

**Comment by filips123** [comment]
Menu categories are determined from [`categories` member](https://www.w3.org/TR/manifest-app-info/#categories-member) of the web app manifest which is mapped to [XDG categories](
 https://specifications.freedesktop.org/menu-spec/menu-spec-1.0.html#category-registry). There's no default category (except `GTK` and `WebApps` which are more "technical" categories) because I don't think a single category would fit different purposes of different web apps.

However, you can specify categories yourself when installing/editing web app in the extension. Adding `network` to that list should then also add desktop entry to the correct menu. There are also [many other](https://github.com/filips123/PWAsForFirefox/blob/main/native/src/integrations/categories.rs#L14-L173) categories you can choose.

**Reply by bkmo** [reply]
Thanks for the input......I did not notice the categories field in the WebApp.

**Comment by prettyv** [comment]
Would it be possible to provide the known XDG categories (or a common subset thereof) as a dropdown/autocomplete for the Categories field in the extension? Ideally not just when none could be mapped from the web app manifest (or none were provided) but also allow adding custom ones to the detected ones?

I don't think most users know offhand which category names are valid and map to which section of their app launchers, I certainly don't. ;)

**Reply by filips123** [reply]
I think autocomplete with known categories could work, I'll try to implement it.
